unstructured/ingest/connector/astra.py

```diff
@@ -111,7 +133,89 @@ def normalize_dict(self, element_dict: dict) -> dict:
         return {
             "$vector": element_dict.pop("embeddings", None),
             "content": element_dict.pop("text", None),
-            "metadata": flatten_dict(
-                element_dict, separator="-", flatten_lists=True, remove_none=True
-            ),
+            "metadata": element_dict,
         }
+
+@dataclass
+class AstraIngestDoc(IngestDocCleanupMixin, BaseSingleIngestDoc):
+    connector_config: SimpleAstraConfig
+    meta: t.Dict[str, str] = field(default_factory=dict)
+    registry_name: str = "astra"
+
+    def update_source_metadata(self):
+        self.source_metadata = SourceMetadata(
+            exists=True,
+        )
+
+
+class AstraSourceConnector(SourceConnectorCleanupMixin, BaseSourceConnector):
+    """Objects of this class support fetching document(s) from Astra DB"""
+
+    connector_config: SimpleAstraConfig
+    _astra_db: t.Optional["AstraDB"] = field(init=False, default=None)
+    _astra_db_collection: t.Optional["AstraDBCollection"] = field(init=False, default=None)
+
+    def to_dict(self, **kwargs):
+        """
+        The _astra_db_collection variable in this dataclass breaks deepcopy due to:
+        TypeError: cannot pickle '_thread.lock' object
+        When serializing, remove it, meaning client data will need to be reinitialized
+        when deserialized
+        """
+        self_cp = copy.copy(self)
+
+        if hasattr(self_cp, "_astra_db_collection"):
+            setattr(self_cp, "_astra_db_collection", None)
+
+        return _asdict(self_cp, **kwargs)
+
+    @property
+    @requires_dependencies(["astrapy"], extras="astra")
+    def astra_db_collection(self) -> "AstraDBCollection":
+        if self._astra_db_collection is not None: ### FIX THIS IS NEVER NONE
```

Comment:
@erichare could you look deeper into this. For some reason `self._astra_db_collection` is never None. It should be when it instantiates. Destination connector part instantiates this as None.

Reply:
@potter-potter this is a strange one for me too since as far as i can tell it's the same logic being used for the destination connector. I'm looking into it